### PR TITLE
Feature: Update wording of our beliefs

### PR DESCRIPTION
**Pull request by rlmoser99**
I noticed the wording of the "Motivation is fueled by working with others" was not quite accurate. It was suggested to reword these into more of a "How it works" section.

## This PR
* Renames the beliefs section to "What you can expect at The Odin Project"'
* Reword each section to encompass similar things that we value and want our learners to be aware of.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -33,20 +33,20 @@ en:
...
         beliefs:
-        - title: Education should be free and accessible
-          description: This curriculum itself is free and we tried to link to resources that are themselves free so anyone in the world can use them.
+        - title: Our lessons build upon each other
+          description: For the best experience, complete all of our lessons and projects in order. Everyone should start with our Foundations path and then complete one of our full-stack paths.
           image_path: about_page/img-education.svg
           image_alt: book icon
-        - title: You learn best by actually building
-          description: The Odin Project curriculum is full of projects that will help you build a strong portfolio of work on GitHub to fill out your resume.
+        - title: Our curriculum is project-based
```
Comment (KevinMulhern) [inline config/locales/en.yml:40]:
```suggestion
        - title: Learn by doing
```

File: config/locales/en.yml [hunk 2]
```diff
@@ -33,20 +33,20 @@ en:
...
           image_alt: keyboard icon
-        - title: Motivation is fueled by working with others
-          description: We're committed to connecting students together so they can stay motivated and learn faster.
+        - title: Receive support from others
+          description: The maintainers of the curriculum run a Discord community, with the help of countless other volunteers, where you can receive help with anything in our curriculum.
           image_path: about_page/img-people.svg
           image_alt: people icon
-        - title: Open source is best
-          description: Our curriculum and website are available on GitHub and we encourage students to actually contribute to the project itself!
+        - title: Learn more by giving back
```
Comment (KevinMulhern) [inline config/locales/en.yml:48]:
```suggestion
        - title: Fully open source
```

File: app/views/static_pages/about.html.erb [hunk 3]
```diff
@@ -10,7 +10,7 @@
         </p>
       </div>
 
-      <h2 class="text-center text-gray-700 dark:text-gray-300 text-2xl mb-12">Our Beliefs</h2>
+      <h2 class="text-center text-gray-700 dark:text-gray-300 text-2xl mb-12">What you can expect at The Odin Project</h2>
```
Comment (KevinMulhern) [inline app/views/static_pages/about.html.erb:13]: ❤️

File: config/locales/en.yml [hunk 4]
```diff
@@ -33,20 +33,20 @@ en:
   static_pages:
     about:
         beliefs:
-        - title: Education should be free and accessible
-          description: This curriculum itself is free and we tried to link to resources that are themselves free so anyone in the world can use them.
+        - title: Our lessons build upon each other
+          description: For the best experience, complete all of our lessons and projects in order. Everyone should start with our Foundations path and then complete one of our full-stack paths.
```
Comment (KevinMulhern) [inline config/locales/en.yml:37]:
```suggestion
      Our free comprehensive curriculum will take you from zero to full stack developer.
```

File: config/locales/en.yml [hunk 5]
```diff
@@ -33,20 +33,20 @@ en:
   static_pages:
     about:
         beliefs:
-        - title: Education should be free and accessible
-          description: This curriculum itself is free and we tried to link to resources that are themselves free so anyone in the world can use them.
+        - title: Our lessons build upon each other
```
Comment (KevinMulhern) [inline config/locales/en.yml:36]: Nit: This one might resonate more if it highlighted the roadmap TOP provides. That is probably our biggest draw for users - getting them out of tutorial hell.

```suggestion
        - title: A full roadmap to becoming a developer
```

**Review by KevinMulhern — COMMENTED** [review]
Thanks for doing this @rlmoser99 , this was **badly** in need of an update 💪 

I think the new content may be leaning a little too much towards how users should use TOP. Tweaking it a little to highlight TOP's best qualities may be better for this pages intended audience - new visitors trying to figure out if this TOP thing for them. It's pretty much a marketing page where we'll have to sell TOP to visitors and thats the kind of language we're best to use.

I've left a few suggestions of how we could maybe do that,  mostly half baked ideas though lol. Just wanted to give you an idea of what I'm getting at.

**Comment by rlmoser99** [comment]
I am going to play around with the wording of the first point and maybe last point.

**Comment by rlmoser99** [comment]
The test that is failing is passing locally and has nothing to do with these changes. 🤔

**Comment by Asartea** [comment]
@rlmoser99 see https://github.com/TheOdinProject/theodinproject/pull/3844#issuecomment-1626107595 and #3924 ; one of the recent changes made that test flaky

**Review by KevinMulhern — APPROVED** [review]
This looks **great** @rlmoser99, nice work! 💪 

Sorry, I feel like I only half explained myself the other day lol. Most of the users who land on this page are from search engines: 

Screenshot 2023-07-10 at 20 52 45

They'll understandably to be looking to find out what they can gain from TOP as quickly as possible after clicking on the search result. This new content will communicate that better than "our beliefs" ever could 🎉

**Pull request merged**